File: original C# files/Section.cs
using System;
using System.Collections.Generic;

namespace ScheduleMaker
{
    public class Section
    {
        //FIELDS
        private int crn;
        private ClassTime lecture;
        private ClassTime labRec;
        private List<ClassTime> classTimes; //both the Lecture object and the LabRec object (if exists) are stored in this list to allow for easy iteration during testing for time conflicts

        //CONSTRUCTORS
        public Section()
        {

        }

        public Section(int crn, ClassTime lecture, ClassTime labRec) //this constructor is used when the section contains a lab or recitation
        {
            this.crn = crn;
            this.lecture = lecture;
            this.labRec = labRec;
            this.classTimes = new List<ClassTime>()
            {
                {lecture},
                {labRec}
            };
        }

        public Section(int crn, ClassTime lecture) //this constructor is used when there is just a lecture
        {
            this.crn = crn;
            this.lecture = lecture;
            this.classTimes = new List<ClassTime>()
            {lecture}; //LabRec is not added to ClassTimes list to avoid NullException during testing
        }

        //ACCESSORS
        public int CRN
        {
            get { return this.crn; }
            set { this.crn = value; }
        }

        public ClassTime Lecture
        {
            get { return this.lecture; }
            set { this.lecture = value; }
        }

        public ClassTime LabRec
        {
            get { return this.labRec; }
            set { this.labRec = value; }
        }

        public List<ClassTime> ClassTimes
        {
            get { return this.classTimes; }
            set { this.classTimes = value; }
        }

        //METHODS
        public bool OnSameDay(Section ASection)
        {
            foreach (ClassTime item in this.classTimes) 
            {
                foreach (ClassTime element in ASection.ClassTimes)
                {
                    if (item.OnSameDay(element)) //call OnSameDay method for each ClassTime object
                    {
                        return true; //if any of the ClassTime objects in ASection have any days in common with any ClassTime objects in this
                    }
                }
            }
            return false; //if the Section object passes the foreach loops, none of the ClassTime objects in it have any days in common
        }

        public bool HasTimeConflict(Section ASection)
        {
            if (!(this.OnSameDay(ASection)))
            {
                return false; //if none of the ClassTime objects share a day, no need to check the times
            }

            foreach (ClassTime item in this.classTimes)
            {
                foreach (ClassTime element in ASection.ClassTimes)
                {
                    if (item.HasTimeConflict(element))
                    {
                        return true; //test all ClssTime objects in each section for time conflicts
                    }
                }
            }
            return false; //if the Section object passes all the foreach loops, then it is compatible
        }
    }
}

File: lambda/generateSchedules/modules/section.js

const ClassTime = require('./classtime');

/**
 * @class
 * Represents a section of a course. Contains a CRN (Course Registration Number), whether or not the section is
 * open, and a list of all ClassTime objects associated with the section. A section can have 1 or more ClassTimes,
 * which can represent either a lecture or lab/recitation component
 */
class Section {
    /**
     * @constructor
     * Creates a new Section
     * @param {[number, string, string, string]} entryInfo - A tuple containing (in the folllowing order):
     * - The CRN (course registration number) for the section
     * - The course name (MATH 1041)
     * - The course title (Calculus I)
     * - The section number (1)
     * @param {[ClassTime]} classtimes - An array of ClassTime objects associated with the section
     * @param {boolean} isOpen - Whether or not the section has any open seats
     * @param {string} campus - The campus where the section takes place
     */
    constructor(entryInfo, classtimes, isOpen, campus){
        [this.crn, this.name, this.title, this.section] = entryInfo;
        this.classtimes = classtimes;
        this.hasLabRec = this.classtimes.length > 1 ? true : false;
        this.isOpen = isOpen;    
        this.campus = campus;              
    }

    /**
     * @method onSameDay - Tests the current Section against another and determines whether 
     * they have any days in common
     * @param {Section} someSection - Another Section object to compare against
     * @return {boolean} TRUE if the sections have any days in common, FALSE otherwise
     */
    onSameDay(someSection){
        //Go through each ClassTime object, and see if they share any days
        return this.classtimes.every(x => 
            someSection.classtimes.every(y => 
                x.onSameDay(y)));
    }

    /**
     * @method hasTimeConflict - Tests the current Section against another and determines whether 
     * they have a time conflict
     * @param {Section} someSection - Another Section object to compare against
     * @return {boolean} TRUE if the sections have a time conflict, FALSE otherwise
     */
    hasTimeConflict(someSection){
        //First test if any days are shared. If not, than no need to test the times
        if(!this.onSameDay(someSection)){
            return false;
        }

        /*Go through each ClassTime object. If any of them have a time conflict, than the section 
        as a whole has a time conflict*/
        return this.classtimes.every(x => 
            someSection.classtimes.every(y => 
                x.hasTimeConflict(y)));
    }
}

/**
 * @exports Section
 */
module.exports = Section;